tests/rgen/Microsoft.Macios.Generator.Tests/BaseGeneratorTestClass.cs

```diff
@@ -16,24 +16,33 @@ namespace Microsoft.Macios.Generator.Tests;
 /// </summary>
 public class BaseGeneratorTestClass {
 	protected BindingSourceGeneratorGenerator GeneratorGenerator;
-	protected CSharpGeneratorDriver _driver;
+	protected CSharpGeneratorDriver Driver;
+
+	// list of the defines for each platform, this is passed to the parser to ensure that
+	// we are testing the platforms as if they were being compiled.
+	readonly Dictionary<ApplePlatform, string []> platformDefines = new () {
+		{ ApplePlatform.iOS, new [] { "__IOS__" } },
+		{ ApplePlatform.TVOS, new [] { "__TVOS__" } },
+		{ ApplePlatform.MacOSX, new [] { "__MACOS__" } },
+		{ ApplePlatform.MacCatalyst, new [] { "__MACCATALYST__" } },
+	};
 
 	public BaseGeneratorTestClass ()
 	{
 		GeneratorGenerator = new BindingSourceGeneratorGenerator ();
-		_driver = CSharpGeneratorDriver.Create (GeneratorGenerator);
+		Driver = CSharpGeneratorDriver.Create (GeneratorGenerator);
 	}
 
 	protected Compilation RunGeneratorsAndUpdateCompilation (Compilation compilation, out ImmutableArray<Diagnostic> diagnostics)
 	{
-		_driver.RunGeneratorsAndUpdateCompilation (compilation, out var updatedCompilation, out diagnostics);
+		Driver.RunGeneratorsAndUpdateCompilation (compilation, out var updatedCompilation, out diagnostics);
 		return updatedCompilation;
 	}
 
 	protected GeneratorDriverRunResult RunGenerators (Compilation compilation)
-		=> _driver.RunGenerators (compilation).GetRunResult ();
+		=> Driver.RunGenerators (compilation).GetRunResult ();
 
-	protected Compilation CreateCompilation (string name, ApplePlatform platform, params string [] sources)
+	protected (Compilation Compilation, ImmutableArray<SyntaxTree> SyntaxTrees) CreateCompilation (string name, ApplePlatform platform, params string [] sources)
```

Comment:
I'm torn about this - 5 years ago, I used tuples as return values frequently, but I found that over time they were harder to maintain than a small structure and without documentation, it was easy to find the method but harder to find out what the meaning of the tuple elements were, especially if they were the same type.

I also notice that you frequently ignore half of the return, so I would be inclined to do this:
```csharp
// return this type
public struct CompilationResult {
    public Compilation Compilation { get; init }
    public ImmutableArray<SyntaxTree> SyntaxTrees { get; init; }
}
// and add:
protected ImmutableArray<SyntaxTree> CreateCompilationSyntaxTrees (...args...) => CreateCompilation (...args...).SyntaxTrees;
protected Compilation CreateCompilationOnly (...args...) => CreateCompilation (...args...).Compilation;
```